rust/src/adapter/mod.rs

```diff
@@ -49,7 +51,7 @@ impl Collector {
         thread::sleep(time::Duration::from_millis(50));
     }
 
-    pub async fn set_metadata(&self, key: String, value: String) {
+    pub async fn set_metadata(&self, key: String, value: u64) {
```

Comment:
I tinkered with a more generic interface for this when doing #15.. I think we should have something like Otel's [KeyValue](https://docs.rs/opentelemetry/0.19.0/opentelemetry/struct.KeyValue.html)